docs/blog/2018-11-29-gatsby-analogy/index.md

```diff
@@ -0,0 +1,103 @@
+
+Take Neo in the Matrix. (If you haven’t seen The Matrix, I’ll give enough context). In the movie, there’s a part where the character Neo has skills uploaded directly into his brain. When he wakes up, he utters, “I know Kung Fu”.
+
+![Neo knows kung fu](./images/neo-kung-fu.gif)
+
+Neo could have spent 10,000 hours mastering the martial art. But instead, thanks to sci-fi software magic, it was instantly uploaded into his brain.
+
+Similarly, you could spend 10,000 hours becoming a kung fu master in performance optimization (and you should, if you want to!). But with Gatsby, you can generate a project, and start off pre-optimized. (A Neo-esque head start.)
+
+**Neo instantly downloads “kung fu”; you instantly download a head start on web performance and optimization.**
+
+## 3. Leverage static site delivery and web app experience
+
+In [Gatsby for apps](/blog/2018-11-07-gatsby-for-apps/), we recently explained:
+
+<pullquote>
+  Gatsby is great for building web experiences that leverage the benefits of both so-called static sites and web applications — simultaneously. You don’t have to sacrifice the advantages of one approach to reap the benefits of the other.
+</pullquote>
+
+When you visit a Gatsby site, a static, server-rendered HTML page is loaded first, and then it hydrates into a React app.
+
+I compare this to modern mattresses that are ordered online, and come compressed in a box, for easier delivery. Once it’s delivered and unboxed, it magically unfurls into a mattress.
+
+![casper mattress inflating](./images/casper-unboxing.gif)
+
+**Both ship small and bloom into much more.**
+
+## 4. Prefetch resources for snappy routing
+
+It’s not just important that Gatsby sites _are_ fast. It’s also important that they _seem_ even faster.
+
+When a Gatsby page finishes loading, Gatsby starts prefetching resources for internal pages the user may navigate to (internal links that are visible in the viewport). It does this at a very low priority -- it will prioritize almost anything else for the current page over these prefetching requests.
+
+On top of that, when a link is actively hovered over (a strong indication the user will visit that link), Gatsby will fetch the resources for that page at a higher priority.
+
+This means that when the user navigates to another internal page, resources have already been quietly loaded in the background and routing feels instantaneous.
+
+Say you order delivery for dinner. It arrives, you eat. Later, you look at the menu again and scan for dessert. You decide on one and press “order”. Your doorbell rings immediately, your dessert has arrived. It’s like the delivery person is a psychic. They arrive instantaneously, with exactly what you wanted, exactly when you want it.
+
+![the flash delivering pizza](./images/the-flash-pizza.gif)
+
+**Gatsby preloads resources for pages a user is likely to navigate to, like a psychic delivery person.**
+
+## 5. Consolidate data sourcing
```

Comment:
And transformations -- which are just as important as sourcing

Also a line I like is "get data from wherever it lives to the right place in your website in the right form exactly when it's needed"

Reply:
tweaked to add 👍